scapy/layers/l2.py

```diff
@@ -805,33 +810,60 @@ def arp_mitm(
         $ sudo scapy
         >>> arp_mitm("192.168.122.156", "192.168.122.17")
 
+    Alternative usages:
+        >>> arp_mitm("10.0.0.1", "10.1.1.0/21", iface="eth1")
+        >>> arp_mitm("10.0.0.1", "10.1.1.2",
+        ...          target_mac="aa:aa:aa:aa:aa:aa",
+        ...          mac2="00:1e:eb:bf:c1:ab")
+
+    .. warning::
+        If using a subnet, this will first perform an arping, unless broadcast is on!
+
     Remember to change the sysctl settings back..
     """
     if not iface:
         iface = conf.route.route(ip1)[0]
     if not target_mac:
         target_mac = get_if_hwaddr(iface)
-    if mac1 is None:
-        mac1 = getmacbyip(ip1)
-        if not mac1:
-            print("Can't resolve mac for %s" % ip1)
-            return
-    if mac2 is None:
-        mac2 = getmacbyip(ip2)
-        if not mac2:
-            print("Can't resolve mac for %s" % ip2)
-            return
-    print("MITM on %s: %s <--> %s <--> %s" % (iface, mac1, target_mac, mac2))
+
+    def _tups(ip, mac):
+        # type: (str, Optional[Union[str, List[str]]]) -> Iterable[Tuple[str, str]]
+        if mac is None:
+            if broadcast:
+                return [(ip, "ff:ff:ff:ff:ff:ff")]
```

Comment:
A bit tricky to understand: this `ip` will be passed as-is (even if it's a range for instance) to `IP`, so the behavior is kept.

Reply:
Maybe you could add that in a comment?